### PR TITLE
Added some comments for ASM hashing

### PR DESCRIPTION
**Pull request by hysz**
## Types of changes

*   [ ] Bug fix (non-breaking change which fixes an issue)
*   [ ] New feature (non-breaking change which adds functionality)
*   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

*   [ ] Change requires a change to the documentation.
*   [ ] Added tests to cover my changes.
*   [ ] Added new entries to the relevant CHANGELOG.jsons.
*   [ ] Labeled this PR with the 'WIP' label if it is a work in progress.
*   [ ] Labeled this PR with the labels corresponding to the changed package.

### CODE REVIEW

File: packages/contracts/src/contracts/current/protocol/Exchange/libs/LibOrder.sol [hunk 1]
```diff
@@ -98,6 +98,26 @@ contract LibOrder is
         bytes32 schemaHash = EIP712_ORDER_SCHEMA_HASH;
         bytes32 makerAssetDataHash = keccak256(order.makerAssetData);
         bytes32 takerAssetDataHash = keccak256(order.takerAssetData);
+        // Assembly for more efficiently computing:
+        // keccak256(abi.encodePacked(
+        //   DOMAIN_SEPARATOR_SCHEMA_HASH,
+        //   keccak256(abi.encodePacked(address(this))),
```
Comment (abandeali1) [inline packages/contracts/src/contracts/current/protocol/Exchange/libs/LibOrder.sol]: Let's drop the `encodePacked` from the single argument hashes.

File: packages/contracts/src/contracts/current/protocol/Exchange/libs/LibOrder.sol [hunk 2]
```diff
@@ -98,6 +98,26 @@ contract LibOrder is
...
+        //   ORDER_SCHEMA_HASH,
+        //   keccak256(abi.encodePacked(
```
Comment (Recmo) [inline packages/contracts/src/contracts/current/protocol/Exchange/libs/LibOrder.sol]: `abi.encode` not `abi.encodePacked` (which would produce a different hash value).

File: packages/contracts/src/contracts/current/protocol/Exchange/libs/LibOrder.sol [hunk 3]
```diff
@@ -98,6 +98,26 @@ contract LibOrder is
         bytes32 schemaHash = EIP712_ORDER_SCHEMA_HASH;
         bytes32 makerAssetDataHash = keccak256(order.makerAssetData);
         bytes32 takerAssetDataHash = keccak256(order.takerAssetData);
+        // Assembly for more efficiently computing:
+        // keccak256(abi.encodePacked(
```
Comment (Recmo) [inline packages/contracts/src/contracts/current/protocol/Exchange/libs/LibOrder.sol]: This function does not compute the outer hash, only the inner one (lines 106-119).

File: packages/contracts/src/contracts/current/protocol/Exchange/libs/LibOrder.sol (hunk 1 above)
Comment (Recmo) [inline packages/contracts/src/contracts/current/protocol/Exchange/libs/LibOrder.sol]: `abi.encode` not `abi.encodePacked` (which would produce a different hash value).

**Pull request merged**